homeassistant/components/geo_location/demo.py

```diff
@@ -0,0 +1,165 @@
+        return entity_id
+
+    def _group_devices(self):
+        """Re-group all entities."""
+        # Sort entries in group by their state attribute (ascending).
+        devices = sorted(self._managed_devices.copy(),
+                         key=lambda device: device.state,
+                         reverse=False)
+        entity_ids = [device.entity_id for device in devices]
+        # Update group.
+        self.group.update_tracked_entity_ids(entity_ids)
+
+    def _generate_random_event(self):
+        """Generate a random event in vicinity of this HA instance."""
+        home_latitude = self._hass.config.latitude
+        home_longitude = self._hass.config.longitude
+
+        # Approx. 111km per degree (north-south).
+        radius_in_degrees = random.random() * MAX_RADIUS_IN_KM / \
+            AVG_KM_PER_DEGREE
+        radius_in_km = radius_in_degrees * AVG_KM_PER_DEGREE
+        angle = random.random() * 2 * pi
+        # Compute coordinates based on radius and angle. Adjust longitude value
+        # based on HA's latitude.
+        latitude = home_latitude + radius_in_degrees * sin(angle)
+        longitude = home_longitude + radius_in_degrees * cos(angle) / \
+            cos(radians(home_latitude))
+
+        event_name = random.choice(EVENT_NAMES)
+        entity_id = self._generate_entity_id(event_name)
+        return DemoGeoLocationEvent(self._hass, entity_id, event_name,
+                                    radius_in_km, latitude, longitude,
+                                    DEFAULT_UNIT_OF_MEASUREMENT)
+
+    def _init_regular_updates(self):
+        """Schedule regular updates based on configured time interval."""
+        track_time_interval(self._hass, lambda now: self._update(),
+                            DEFAULT_UPDATE_INTERVAL)
+
+    def _update(self, count=1):
+        """Remove events and add new random events."""
+        # Remove devices.
+        for _ in range(1, count + 1):
+            if self._managed_devices:
+                device = random.choice(self._managed_devices)
+                if device:
+                    _LOGGER.debug("Removing %s", device)
+                    self._managed_devices.remove(device)
+                    self._hass.add_job(device.async_remove())
+        # Generate new devices from events.
+        new_devices = []
+        for _ in range(1, count + 1):
+            new_device = self._generate_random_event()
+            _LOGGER.debug("Adding %s", new_device)
+            new_devices.append(new_device)
+            self._managed_devices.append(new_device)
+        self._add_devices(new_devices)
+        self._group_devices()
+
+
+class DemoGeoLocationEvent(GeoLocationEvent):
+    """This represents a demo geo location event."""
+
+    def __init__(self, hass, entity_id, name, distance, latitude, longitude,
+                 unit_of_measurement):
+        """Initialize entity with data provided."""
+        super().__init__()
```

Comment:
There are no init methods in the parent classes.

Reply:
OK